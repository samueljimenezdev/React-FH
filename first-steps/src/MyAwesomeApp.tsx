import type { CSSProperties } from "react"

// Se colocan afuera las constantes para no gastar recursos en el render del componente
const firstName = 'Samuel'
const lastName = 'Jimenez'
const myStyles: CSSProperties = {
    backgroundColor: '#FAFAFA',
    borderRadius: 20,
    padding: 10,
    marginTop: 30,
}
export function MyAwesomeApp() {

    return (
        <>
            <h1>{firstName}</h1>
            <h3 style={myStyles}>{lastName}</h3>
        </>
    )
}