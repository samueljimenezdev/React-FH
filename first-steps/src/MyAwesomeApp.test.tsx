import { describe, expect, test } from "vitest";
import { MyAwesomeApp } from "./MyAwesomeApp";
import { render, screen } from '@testing-library/react'

describe('My Awesome App', () => {
    test('should render firstName and lastName', () => {
        const { container } = render(<MyAwesomeApp />); // Evalua estados iniciales, no cambios, para eso usar screen()

        const h1 = container.querySelector('h1');
        expect(h1?.innerHTML).toContain('Samuel'); // Contener

        const h3 = container.querySelector('h3');
        expect(h3?.innerHTML).toBe('Jimenez'); // Ser primitivos

    })


    test('should render firstName and lastName using screen()', () => {
        render(<MyAwesomeApp />); // Evalua estados iniciales, no cambios, para eso usar screen()
        screen.debug()

        const h1 = screen.getByTestId('fist-name-title')
        expect(h1.innerHTML).toContain('Samuel')
    })

    test('should match snapshot', () => {
        const { container } = render(<MyAwesomeApp />)
        expect(container).toMatchSnapshot(); // Cuando se ejecuta crea un snapshot en la carpeta __snapshots__
    })
})