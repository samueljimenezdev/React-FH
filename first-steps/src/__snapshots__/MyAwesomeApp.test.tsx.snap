// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`My Awesome App > should match snapshot 1`] = `
<div>
  <div>
    <h1
      data-testid="fist-name-title"
    >
      Samuel
    </h1>
    <h3
      style="background-color: rgb(250, 250, 250); border-radius: 20px; padding: 10px; margin-top: 30px;"
    >
      Jimenez
    </h3>
    <p
      class="label-salduo"
    >
      Hola mundo
    </p>
  </div>
</div>
`;
